### ISSUE TITLE
HolBdd and temporal_deep examples cannot build under Moscow ML

### ISSUE THREAD

**Issue by binghe** [HOL-Theorem-Prover/HOL#769]
Hi,

I believe this issue was brought by the multidir Holmake feature.   In `examples/HolBdd/Holmakefile`, there's a `INCLUDE` to `examples/muddy`:

```
INCLUDES = ../muddy
```

In `examples/muddy` there's `Holmakefile` at all, instead there are just some `*.sig` and `*.sml` files.   However, when `Holmake` arrives at this folder, somehow it fails with the following strange error saying "bvec.uo seems to depend on itself": 

```
*** Switching to execute /Users/binghe/ML/HOL.mosml/bin/Holmake
*** (As we are in/under its HOLDIR)
Holmake: Scanning ../muddy
Holmake: /Users/binghe/ML/HOL.mosml/examples/muddy/bvec.uo seems to depend on itself failing
 Loop is : /Users/binghe/ML/HOL.mosml/examples/muddy/bvec.uo>/Users/binghe/ML/HOL.mosml/examples/muddy/bvec.uo
```

Due to the same reason, `temporal_deep` also cannot build under Moscow ML, because `${HOLDIR}/examples/muddy` is included  in `examples/temporal_deep/src/model_check/Holmakefile`.

--Chun

**Comment by mn200** [comment]
The issue is that Holmake's dependency analysis thinks that `open bdd fdd prim_type bvec` induces a dependency on `bvec`.  But under Moscow ML, `prim_type` is a keyword, and the scope of the `open` should end just before it.

**Comment by mn200** [comment]
I think this should be fixed by e85ed9b48.

**Comment by binghe** [comment]
Amazing ... never thought the solution is in that way:) I confirm now it works. Issue closed. Thanks!
